### ISSUE TITLE
several accessibility bugs in layout

### ISSUE THREAD

**Issue by tschiela** [google/material-design-lite#1179]
MDL version: 1.0.0
Browser: Chrome 
Browser version: 43
Operating system: Ubuntu 
Operating system version: 14.04
URL: 
http://www.getmdl.io/templates/android-dot-com/index.html
http://www.getmdl.io/components/index.html#layout-section

What steps will reproduce the problem:
1. open page
2. press [TAB]
3. navigation should be closed by pressing [ESC]

What is the expected result?
drawer button will be focused. With a press on [Enter] the drawer navigation will be opened.

What happens instead of that?
Another element will be focused. E.g. in the Android template the search input field will be focused. Here seems to be the next glitch. The links in the horizontal navigation will be focused after the search field. So the element order not match witch the tab order. 

My proposal for a solution:
- the .mdl-layout__drawer-button should be a button, not a div
- add missing aria attributes (aria-expanded to the button, aria-hidden to the navigatio. Maybe aria-owns to the button, iam not shure about that)
- drawer navigation items get tabindex of -1, if drawer nav is closed and 0 if drawer nav is open
- the html elements of the navigation elements (links and search input) should be in the right order in the markup and not be positioned by to many css magic (order change with flexbox)
- close navigation on pressing [ESC]

**Comment by tschiela** [comment]
What is the state of the bug? I could create an PR for that, if this is desired.

**Comment by surma** [comment]
Hey @tschiela,

sorry for the missing response. Nothing has been done yet, but it will be addressed once we get around to it. However, PRs are greatly appreciated. So if you want to take care of this, feel free to go ahead :)
